pkg/controller/baremetalhost/baremetalhost_controller.go

```diff
@@ -385,6 +386,39 @@ func (r *ReconcileBareMetalHost) Reconcile(request reconcile.Request) (reconcile
 		}
 	}
 
+	// Check the current power status against the desired power
+	// status.
+	if provResult, err = prov.UpdateHardwareState(); err != nil {
+		return reconcile.Result{}, errors.Wrap(err, "failed to update the hardware status")
+	}
+	if provResult.Dirty {
+		if err := r.saveStatus(host); err != nil {
+			return reconcile.Result{}, errors.Wrap(err,
+				"failed to update host hardware state")
+		}
+		return reconcile.Result{RequeueAfter: provResult.RequeueAfter}, nil
+	}
+	if host.Status.PoweredOn != host.Spec.Online {
```

Comment:
If I grok this correctly. If it is not powered on, and we expect it ?deployed? then proceed?

Reply:
If the desired power state (in the spec) is not the same as the actual power state (in the status) then we want to change the state.